### PR TITLE
add option for preview navigation behavior

### LINKED ISSUE #112348: Ability to switch to previous behavior on 1.52.0 'Preview editor improvements'
With the November 2020 update, code navigation options do not open the navigated file in the same editor even when the current editor is in the preview mode. This quickly litters the workbench with a ton of editors when I want to drill down inheritance trees for example. If I want the navigation target to open in a new editor I can quickly get out of the preview mode with a command like `save` and then navigate so I don't understand the need for this change. I was able to use the `Go->Back` command to go to the previous editor in the preview mode already so I never lost track of where I have navigated from and this way the workbench is not littered with a ton of editors I'm not interested in. It really goes against the nature of "preview mode" in my opinion.

So my request it, please make it toggleable, I want the old behavior back. By old behavior I mean if the current editor is in preview mode and I navigate from it with commands like go to definition, open the definition file in the same editor in preview mode.

### PR DESCRIPTION
**Pull request by goldst**
This PR fixes #112348
Let me know whether the description text of the new option is ok, so I can fix it if necessary.

### CODE REVIEW

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 1]
```diff
@@ -133,6 +133,11 @@ import { isStandalone } from 'vs/base/browser/browser';
 				'description': nls.localize('enablePreviewFromQuickOpen', "Controls whether editors opened from Quick Open show as preview. Preview editors do not keep open and are reused until explicitly set to be kept open (e.g. via double click or editing)."),
 				'default': false
 			},
+			'workbench.editor.enablePreviewFromCodeNavigation': {
+				'type': 'boolean',
+				'description': nls.localize('enablePreviewFromCodeNavigation', "Controls whether preview editors should be kept open as regular editors when opening a second editor from the preview (e.g. via go to definition)"),
+				'default': true
```
Comment (bpasero) [inline src/vs/workbench/browser/workbench.contribution.ts:139]:
```suggestion
				'default': false
```

File: src/vs/workbench/services/editor/browser/codeEditorService.ts [hunk 2]
```diff
@@ -75,7 +77,9 @@ export class CodeEditorService extends CodeEditorServiceImpl {
 		// should be pinned or not. This ensures that the source of a navigation
 		// is not being replaced by the target. An example is "Goto definition"
 		// that otherwise would replace the editor everytime the user navigates.
+		const pinPreview = this.configurationService.getValue<boolean>('workbench.editor.enablePreviewFromCodeNavigation');
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/codeEditorService.ts:80]: I would prefer if we would lookup this setting similar to how we do it here:

https://github.com/microsoft/vscode/blob/4031280b29ee6e6d85abac0da91167440c575175/src/vs/workbench/contrib/codeEditor/browser/quickaccess/gotoLineQuickAccess.ts#L33-L33

Aka add it to `IWorkbenchEditorConfiguration`

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/workbench.contribution.ts (hunk 1 above)
Comment (goldst) [reply src/vs/workbench/browser/workbench.contribution.ts:139]: Are you sure? In my implementation `true` means new behavior (keeping old preview editor open). Your suggestion would make the old behavior default again.  I can also change it to `false` means new behavior, if that's what you mean.

File: src/vs/workbench/services/editor/browser/codeEditorService.ts (hunk 2 above)
Comment (goldst) [reply src/vs/workbench/services/editor/browser/codeEditorService.ts]: changed it, thanks for the pointer

**Comment by goldst** [comment]
@bpasero, would you please help me with the macOS Integration Tests? I'm not sure why it fails all of a sudden, I don't think I changed anything that could cause a timeout... is there a way to restart it?

**Comment by bpasero** [comment]
Thanks 🥂

**Pull request merged**